t/lib/Common.pm

```diff
@@ -62,12 +65,20 @@ sub skip_unless_has_secret {
     plan skip_all => 'PERL_STRIPE_TEST_API_KEY is required' unless api_key();
 }
 
+memoize 'stripe';
 sub stripe {
     my %params = @_;
+    my $no_mock = delete($params{no_mock}) || api_key();
+    if ($ENV{LWP_RECORD} or !$no_mock) {
+        $params{ua} = Test::LWP::Recorder->new({
+            record    => $ENV{LWP_RECORD},
+            cache_dir => 't/LWPCache/' . basename($0),
+        });
+    }
     state $client = WebService::Stripe->new(
         api_key => api_key(),
         version => '2014-12-17',
-        %params
+        %params,
     );
     return $client;
 }
```

Comment:
I really don't like `no_`-prefixed variables. Affirmative variables (e.g. `mock_lwp => 1`) tend to be clearer / less confusing. Suggested rewrite:

```perl
memoize 'stripe';
sub stripe {
    my %params = @_;
    $params{'api_key'}         //= api_key;
    $params{'version'}         //= '2014-12-17';
    $params{'mock_lwp_record'} //= $ENV{'PERL_STRIPE_MOCK_RECORD'};
    $params{'mock_lwp'}        //= $ENV{'PERL_STRIPE_MOCK'}
        // !$params{'api_key'};

    my %client;
    $client{'api_key'} = $params{'api_key'};
    $client{'version'} = $params{'version'};
    $client{'ua'}      = Test::LWP::Recorder->new({
        record    => $params{'mock_lwp_record'},
        cache_dir => 't/LWPCache/' . basename($0),
    }) if $params{'mock_lwp'};

    state $client = WebService::Stripe->new(%client);
    return $client;
}
```